### PR TITLE
docs(v2): Restyle migration guide box

### PR DESCRIPTION
**Pull request by cjquines**
* make the box collapsible (this is what i'm least sure of)
* make the headings smaller
* make the code blocks wider

image

### CODE REVIEW

**Comment by cjquines** [comment]
could also add some function tag to mark stuff that had breaking changes

File: docs/src/components/v1/migration-box.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import type { ReactNode } from "react";
+import { Button } from "@/components/ui/button";
+import {
+  CollapsibleContent,
+  Collapsible as CollapsibleRoot,
```
Comment (eranhirsch) [inline docs/src/components/v1/migration-box.tsx:5]: Is there a reason we need this rename?
```suggestion
  Collapsible,
```

**Review by eranhirsch — APPROVED** [review]
I think that if we go with this approach we can drop the "No changes" blocks for functions with migration notes, and maybe change the navigation label for the v1 docs to something like "Docs (Legacy, V1)" or something else that makes sense.

File: docs/src/components/markdown.astro [hunk 2]
```diff
@@ -3,14 +3,13 @@ import { asHtml } from "@/lib/as-html";
...
-const { markdown } = Astro.props;
+const { markdown, className } = Astro.props;
 ---
 
 <article
-  class="prose prose-slate dark:prose-invert prose-code:text-primary prose-code:before:content-none prose-code:after:content-none"
+  class={`prose prose-slate dark:prose-invert prose-code:text-primary prose-code:before:content-none prose-code:after:content-none ${className ?? ""}`}
   set:html={markdown === undefined ? undefined : asHtml(markdown)}
->
-  <slot />
```
Comment (eranhirsch) [inline docs/src/components/markdown.astro:15]: I'm confused, how does this work if you removed the slot? We use the slotted approach for the migration boxes...

**Review by eranhirsch — APPROVED** [review]
Thanks for addressing my comments.

File: docs/src/components/markdown.astro (hunk 2 above)
Comment (cjquines) [reply docs/src/components/markdown.astro:15]: `set:html` is the thing that does this here; this was apparently against the https://ota-meshi.github.io/eslint-plugin-astro/rules/no-conflict-set-directives/ rule

Comment (eranhirsch) [reply docs/src/components/markdown.astro:15]: but doesn't this mean we need to send the markdown as a param instead of a child in the migration box?

Comment (cjquines) [reply docs/src/components/markdown.astro:15]: huh yeah, i checked and i guess they're both needed????

**Pull request merged**